src/types.ts

```diff
@@ -39,9 +39,7 @@ export type IPluginHook = (ad: IAd) => void;
 
 export type IPluginConstructorOrSingleton = IPluginConstructor | IPlugin;
 
-export interface IPluginConstructor {
-  new(ad: IAd): IPlugin;
-}
+export type IPluginConstructor = new(ad: IAd) => IPlugin;
```

Comment:
Is it just for consistency that you are naming simple types with an `I` prefix?
Usually that is used in typescript to signify an `interface`.

Reply:
I didn't actually do this!